### PR TITLE
Support extracting relay info from the ledger state

### PR DESCRIPTION
**Pull request by karknu**
A relay can either be a domain name or an IP address therefor
DomainAddress isn't suitable.

### CODE REVIEW

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/PeerSelection.hs [hunk 1]
```diff
@@ -59,26 +61,35 @@ instance LedgerSupportsPeerSelection (ShelleyBlock era) where
...
               $ shelleyLedgerState
 
-      relayToDomainAddress :: SL.StakePoolRelay -> DomainAddress
-      relayToDomainAddress = undefined
+      relayToRelayAddress :: SL.StakePoolRelay -> Maybe RelayAddress
+      relayToRelayAddress (SL.SingleHostAddr (SJust (Port port)) (SJust ipv4) _       ) =
+          Just $ RelayAddressAddr (IPv4 ipv4) (fromIntegral port)
+      relayToRelayAddress (SL.SingleHostAddr (SJust (Port port)) SNothing (SJust ipv6)) =
+          Just $ RelayAddressAddr (IPv6 ipv6) (fromIntegral port)
+      relayToRelayAddress (SL.SingleHostName (SJust (Port port)) dnsName)     =
```
Comment (mrBliss) [inline ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/PeerSelection.hs:69]: Either align or don't align, not a mix of the two :slightly_frowning_face: 
```suggestion
      relayToRelayAddress (SL.SingleHostAddr (SJust (Port port)) (SJust ipv4) _) =
          Just $ RelayAddressAddr (IPv4 ipv4) (fromIntegral port)
      relayToRelayAddress (SL.SingleHostAddr (SJust (Port port)) SNothing (SJust ipv6)) =
          Just $ RelayAddressAddr (IPv6 ipv6) (fromIntegral port)
      relayToRelayAddress (SL.SingleHostName (SJust (Port port)) dnsName) =
```

**Review by mrBliss — APPROVED** [review]
LGTM, except for the strange alignment.

**Comment by karknu** [comment]
bors +r

**Pull request merged**